internal/token/migration.go

```diff
@@ -0,0 +1,140 @@
+package token
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/ubuntu/authd-oidc-brokers/internal/fileutils"
+	"golang.org/x/crypto/scrypt"
+)
+
+const saltLen = 32
+
+// UseOldEncryptedToken checks if the password file or the old encrypted token file exists. It returns false if the
+// password file exists, true if only the old encrypted token file exists, and an error if neither exists.
+func UseOldEncryptedToken(passwordPath, oldEncryptedTokenPath string) (bool, error) {
+	exists, err := fileutils.FileExists(passwordPath)
+	if err != nil {
+		return false, fmt.Errorf("could not check password file: %w", err)
+	}
+	if exists {
+		return false, nil
+	}
```

Comment:
Note to self: This causes things to break if the password file exists but the token does not.

Reply:
I fixed that yesterday